public/app/features/dashboard-scene/utils/DashboardModelCompatibilityWrapper.ts

```diff
@@ -175,39 +166,7 @@ export class DashboardModelCompatibilityWrapper {
       return;
     }
 
-    const gridItem = vizPanel.parent;
-    if (!(gridItem instanceof DashboardGridItem)) {
-      console.error('Trying to remove a panel that is not wrapped in DashboardGridItem');
-      return;
-    }
-
-    const layout = sceneGraph.getLayout(vizPanel);
-    if (!(layout instanceof SceneGridLayout)) {
-      console.error('Trying to remove a panel in a layout that is not SceneGridLayout ');
-      return;
-    }
-
-    // if grid item is directly in the layout just remove it
-    if (layout === gridItem.parent) {
-      layout.setState({
-        children: layout.state.children.filter((child) => child !== gridItem),
-      });
-    }
-
-    // Removing from a row is a bit more complicated
-    if (gridItem.parent instanceof SceneGridRow) {
-      // Clone the row and remove the grid item
-      const newRow = layout.clone({
-        children: layout.state.children.filter((child) => child !== gridItem),
-      });
-
-      // Now update the grid layout and replace the row with the updated one
-      if (layout.parent instanceof SceneGridLayout) {
-        layout.parent.setState({
-          children: layout.parent.state.children.map((child) => (child === layout ? newRow : child)),
-        });
-      }
-    }
+    this._scene.removePanel(vizPanel);
```

Comment:
One thing I would consider here is to not create layout modifying method on the dashboard scene, but instead have a heper function that resolves `DashboardLayoutManager` for a `DashboardScene` and one would use manager's interfice directly:

`getLayoutManager(scene).removePanel(vizPanel)`

This way we don't have to maintaint two mnipulation interfaces for the same thing.

Reply:
agree generally, the DashboardScene addPanel does "generic" (layout agnostic) check if the dashboard is in edit mode or not and goes into edit mode, removePanel might also have some dashboard level state logic (like updating isEmpty state).

But depending on how those change in the future I think we should be able to get rid of most of these dashboard level layout manipulation functions